packages/browser/src/transports/xhr.ts

```diff
@@ -20,6 +20,26 @@ export class XHRTransport extends BaseTransport {
       });
     }
 
+    let payload = JSON.stringify(event);
+    let url = this.url;
+    let headers = this.options.headers || {};
+
+    if (event.type === 'transaction') {
+      url = url.replace('/store', '/envelope');
+      headers['content-type'] = 'application/x-sentry-envelope';
+
+      const envelopeHeaders = JSON.stringify({
+        event_id: event.event_id,
+        sent_at: null,
+      });
+      const itemHeaders = JSON.stringify({
+        content_type: 'application/json',
+        type: event.type,
+      });
+      const envelope = `${envelopeHeaders}\n${itemHeaders}\n${payload}\n`;
+      payload = envelope;
```

Comment:
Can we extract that part into the utils package and make a function out of it?
It's duplicated in a few places.